source/generators/Back2backGammas.cc

```diff
@@ -100,9 +100,10 @@ namespace nexus {
 
     if (costheta_min_ != -1. || costheta_max_ != 1. || phi_min_ != 0. || phi_max_ != 2.*pi) {
       G4ThreeVector p = nexus::Direction(costheta_min_, costheta_max_, phi_min_, phi_max_);
-      px = p.x() * pmod;
-      py = p.y() * pmod;
-      pz = p.z() * pmod;
+      p = p * pmod;
+      px = p.x();
+      py = p.y();
+      pz = p.z();
```

Comment:
As I suggested in my PR:

https://github.com/next-exp/nexus/blob/6829fcfc397f405808f21e746339bd873290b9b1/source/generators/Back2backGammas.cc#L103-L119
Create the scaled vector in one go, and use the accessors at the point where you need the components.

If you don't like the ternary operator, default construct `momentum_direction` beforehand, and overwrite in an `if-then-else` that replaces the ` ? : `.

Reply:
Given that we both seem to be at this right now, how about con-peras?

Reply:
That code sample above contains too much stuff, so let's break it down. First: https://github.com/next-exp/nexus/blob/6829fcfc397f405808f21e746339bd873290b9b1/source/generators/Back2backGammas.cc#L107

Then: https://github.com/next-exp/nexus/blob/6829fcfc397f405808f21e746339bd873290b9b1/source/generators/Back2backGammas.cc#L113-L114

And also note that all the setting of particle properties (absent from my version) is pointless, because the particle *already has* those properties with those values!

Reply:
Done